ambry-frontend/src/main/java/com/github/ambry/frontend/UserQuotaRequestCostPolicy.java

```diff
@@ -66,28 +66,11 @@ private String restMethodToCostMetric(RestMethod restMethod) {
 
   /**
    * Calculate the storage cost incurred to serve a request.
-   * For post requests this is the number bytes in GB of the blob data. For all other write requests the default is DEFAULT_COST_FOR_HEAD_DELETE_TTL.
-   * For read requests there is no storage cost incurred.
    * @param restRequest {@link RestRequest} to find type of request.
-   * @param restResponseChannel {@link RestResponseChannel} object.
    * @return storage cost.
    */
-  private double calculateStorageCost(RestRequest restRequest, RestResponseChannel restResponseChannel) {
-    switch (restRequest.getRestMethod()) {
-      case POST:
-        long contentSize = 0;
-        if (restRequest.getUri().contains(Operations.STITCH)) {
-          contentSize = Long.parseLong((String) restResponseChannel.getHeader(RestUtils.Headers.BLOB_SIZE));
-        } else {
-          contentSize = restRequest.getBytesReceived();
-        }
-        return contentSize / BYTES_IN_GB;
-      case DELETE:
-      case PUT:
-        return CU_COST_UNIT / BYTES_IN_GB; // Assuming 1 chunk worth of storage cost for deletes and ttl updates.
-      default:
-        return 0;
-    }
+  private double calculateStorageCost(RestRequest restRequest) {
+    return RestUtils.isUploadRequest(restRequest) ? restRequest.getBlobBytesReceived() / BYTES_IN_GB : 0;
```

Comment:
I think this cost calculation logic needs more thought for stitched requests. Also we are charging customers for ttl updates and deletes as well.

Reply:
same above.

Reply:
Right now the contract is that any request that comes to Ambry storage node, needs to get charged with a min storage cost which is same as if for a 4MB chunk. The same logic is also used by ResourceIntelligence when we send them the cost values. If we want to change this behavior we should talk with Rob if he has any concerns about this.

Reply:
I don't want to miss use this info here, if this storage cost is shared for RI, then I should probably ignore this cost in my enforcer implementation. I can get the blob size from restRequest when charging storage quota. i don't need cost map passed to my enforcer.